apiserver/params/network_test.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package params_test
+
+import (
+	"encoding/json"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/network"
+)
+
+type (
+	P struct {
+		prot string
+		num  int
+	}
+	IS []interface{}
+	M  map[string]interface{}
+)
+
+type NetworkSuite struct{}
+
+var _ = gc.Suite(&NetworkSuite{})
+
+func (s *NetworkSuite) TestPortsResults(c *gc.C) {
+	// Convenience helpers.
+	mkPortsResults := func(prs ...params.PortsResult) params.PortsResults {
+		return params.PortsResults{
+			Results: prs,
+		}
+	}
+	mkPortsResult := func(msg, code string, ports ...P) params.PortsResult {
+		pr := params.PortsResult{}
+		if msg != "" {
+			pr.Error = &params.Error{msg, code}
+		}
+		for _, p := range ports {
+			pr.Ports = append(pr.Ports, params.Port{p.prot, p.num})
+		}
+		return pr
+	}
+	mkResults := func(rs ...interface{}) M {
+		return M{"Results": rs}
+	}
+	mkResult := func(err, ports interface{}) M {
+		return M{"Error": err, "Ports": ports}
+	}
+	mkError := func(msg, code string) M {
+		return M{"Message": msg, "Code": code}
+	}
+	mkPort := func(prot string, num int) M {
+		return M{"Protocol": prot, "Number": num}
+	}
+	// Tests.
+	tests := []struct {
```

Comment:
Tried now to find a way it's more compact but the structures are still good readable.